packages/astro/src/assets/utils/transformToPath.ts

```diff
@@ -10,7 +10,7 @@ export function propsToFilename(transform: ImageTransform, hash: string) {
 		isESMImportedImage(transform.src) ? transform.src.src : transform.src
 	);
 	const ext = extname(filename);
-	filename = basename(filename, ext);
+	filename = decodeURIComponent(basename(filename, ext));
```

Comment:
Hmm, while this does fix the problem I'm sure, I wonder what kind of filepath does it generate? Doesn't that mean that for non-encoded slashes it would generate nested directories?

Reply:
A non-encoded slash would be removed in the `basename()` operation.

Reply:
Oh yeah, that's true. Awesome then!